helix-view/src/editor.rs

```diff
@@ -398,18 +412,88 @@ impl Editor {
             .selections
             .remove(&id);
 
-        if close_buffer {
-            // get around borrowck issues
-            let doc = &self.documents[&view.doc];
+        self.tree.remove(id);
+        self._refresh();
+    }
+
+    pub fn close_document(&mut self, doc_id: DocumentId, force: bool) -> anyhow::Result<()> {
+        if !self.documents.contains_key(&doc_id) {
+            anyhow::bail!("document does not exist");
+        }
 
-            if let Some(language_server) = doc.language_server() {
-                tokio::spawn(language_server.text_document_did_close(doc.identifier()));
+        let doc = &self.documents[&doc_id];
```

Comment:
Rather than doing essentially two lookups, you can use `.get`:

```rust
let doc = match self.documents.get(&doc_id) {
   Some(doc) => doc,
   None => anyhow::bail!("document does not exist"),
};
```